packages/components/modules/navigations/Header/AccountMenu/AccountPopover/index.tsx

```diff
@@ -1,81 +1,135 @@
 'use client'
 
-import { FC } from 'react'
+import { FC, Fragment } from 'react'
 
 import { User as BaseUser, useJWTUser, useLogout } from '@baseapp-frontend/authentication'
-import { ClickableAvatar, Popover, usePopover } from '@baseapp-frontend/design-system'
-import { JWTContent, joinWithSeparator } from '@baseapp-frontend/utils'
+import { Popover, usePopover } from '@baseapp-frontend/design-system'
+import { JWTContent } from '@baseapp-frontend/utils'
 
+import { ButtonBase, List } from '@mui/material'
 import Box from '@mui/material/Box'
 import Divider from '@mui/material/Divider'
 import MenuItem from '@mui/material/MenuItem'
 import Stack from '@mui/material/Stack'
-import Typography from '@mui/material/Typography'
 
-import { AccountPopoverProps } from './types'
+import AccountAvatar from '../AccountAvatar'
+import CurrentUserPlaceholder from '../CurrentUserPlaceholder'
+import { PopoverStyles as DefaultPopoverStyles } from './styled'
+import { AccountPopoverProps, ComponentItems } from './types'
 
 const AccountPopover: FC<AccountPopoverProps> = ({
+  accountAvatarUrl,
+  onCloseCallback,
   menuItems = [],
+  PopoverStyles = {},
+  accountInfoSectionController = { show: true, items: [] },
+  menuItemsSectionController = { show: true, items: [] },
+  menuActionsSectionController = { show: true, items: [] },
+  extraSectionControllersList = [],
   logoutButtonLabel = 'Logout',
+  disableCurrentUserPlaceholder = false,
   hideLogoutButton = false,
 }) => {
   const { user } = useJWTUser<BaseUser & JWTContent>()
   const { logout } = useLogout()
+
   const popover = usePopover()
 
-  const handleLogout = async () => {
+  const handleOnClose = () => {
+    if (onCloseCallback) {
+      onCloseCallback()
+    }
     popover.onClose()
+  }
+
+  const handleLogout = async () => {
+    handleOnClose()
     logout()
   }
 
   const handleMenuItemClick = (onClick: () => void) => {
     onClick()
-    popover.onClose()
+    handleOnClose()
   }
 
+  const renderItems = (items: ComponentItems) =>
+    items.map((item, index) => {
+      if (!item) {
+        return null
+      }
+      if (typeof item === 'function') {
+        const Component = item as FC
+        return <Component key={index} /> // eslint-disable-line react/no-array-index-key
+      }
+      return <Fragment key={index}>{item}</Fragment> // eslint-disable-line react/no-array-index-key
+    })
+
   return (
     <>
-      <ClickableAvatar
-        src={user?.avatar?.small}
-        alt={user?.email}
-        onClick={popover.onOpen}
-        isOpen={!!popover.open}
+      <AccountAvatar
+        src={accountAvatarUrl ?? user?.avatar?.small}
+        alt="User avatar"
+        popoverOpen={Boolean(popover.open)}
+        popoverOnOpen={popover.onOpen}
+      />
+
+      <Popover
+        open={popover.open}
+        onClose={handleOnClose}
+        sx={{ ...DefaultPopoverStyles, ...PopoverStyles }}
       >
```

Comment:
I got the idea behind the controllers, but it looks over complex to me. We could move these sections into components and make them overridable by the template. I think we should do something like:

```
const AccountPopover: FC<AccountPopoverProps> = ({
  // ...
  CurrentUser = DefaultCurrentUser,
  SwitchProfileMenu = DefaultSwitchProfileMenu,
  MenuItems = DefaultMenuItems,
  LogoutItem = DefaultLogoutItem,
}) => {
  // ...
  return (
    <>
      <ClickableAvatar
        src={user?.avatar?.small}
        alt={user?.email}
        onClick={popover.onOpen}
        isOpen={!!popover.open}
      >
        {user?.firstName && user?.lastName ? `${user?.firstName[0]}${user?.lastName[0]}` : ''}
      </ClickableAvatar>
      <Popover
        open={popover.open}
        onClose={handleOnClose}
        sx={{ ...DefaultPopoverStyles, ...PopoverStyles }}
      >
        <CurrentProfile /> // why name it a CurrentUserPlaceholder ?
        <SwitchProfileMenu /> // make sure that's an actual component

        {!!CurrentUser && !!SwitchProfileMenu && <Divider sx={{ borderStyle: 'solid' }} />}
        
        <MenuItems menuItems={menuItems} /> // create this new component rendering the menu items, it should accept a menuItems prop

        {!!LogoutItem && <Divider sx={{ borderStyle: 'solid' }} />}

        <LogoutItem /> // move logout logic inside this component
      </Popover>
    </>
  )
}
```

it think that should provide already a decent customization option for a v1. If we wanna provide a couple more granular customization options, we could expose these component props, like:

```
const AccountPopover: FC<AccountPopoverProps> = ({
  // ...
  SwitchProfileMenu = DefaultSwitchProfileMenu,
  SwitchProfileMenuProps = {},
  LogoutItem = DefaultLogoutItem,
  LogoutItemProps = {}
}) => {
  // ...
  return (
    <>
      <Popover>
        // ...
        <SwitchProfileMenu {...SwitchProfileMenuProps} />

        <LogoutItem {...LogoutItemProps} />
      </Popover>
    </>
  )
}
```

then let's say on the template, we don't want any of these profile features, we could just do:
```
    <BaseNavigationLayout
       // ...
      AccountMenuProps={{
        // ...
        AccountPopoverProps: {
          CurrentProfile: null,
          SwitchProfileMenu: null
        }
      }}
    >
      {children}
    </BaseNavigationLayout>
```

Reply:
I wanted to reuse the component to avoid redundancy. I also didn't want to break compatibility with the previous version of it. I don't want to mix the domains of profiles and navigations. By using the controller, I didn't break compatibility, and one day, if I (or any developer with their respective projects) decide to remove the profiles component from the project and use the simple AccountPopover, they can do so easily.

I understand that we want to avoid overcomplicating components, but the only other way here is to have redundant code. I chose to make it more complex. However, if you think it's better, I can make it redundant. But I truly believe we shouldn't mix the domains, since I already avoided that.